File: test/e2e/test-files/workspaces/generate-data-frames-r/generate-data-frames.r
library(tidyverse)
library(arrow)

# seems like a reasonable upper limit on an m3 with 36Gb
num_rows <- 30000
num_cols <- 30000

# Create a data frame with random integers
data <- replicate(num_cols, sample(0:99, num_rows, replace = TRUE))

# Convert the matrix to a data frame and assign column names
df <- as.data.frame(data)
colnames(df) <- paste0('col', 1:num_cols)

file_path <- file.path(getwd(), 'data-files', '30kX30k.parquet')

# don't add to github:
write_parquet(df, file_path)

File: test/e2e/test-files/workspaces/generate-data-frames-py/generate-data-frames.py
import pandas as pd
import numpy as np
import os

# seems like a reasonable upper limit on an m3 with 36Gb
num_rows = 80000
num_cols = 80000

data = {}
for i in range(num_cols):
    col_name = f'col{i+1}'
    data[col_name] = np.random.randint(low=0, high=100, size=num_rows)

df = pd.DataFrame(data)

# don't add to github:
filePath = os.path.join(os.getcwd(), 'data-files', '80kX80k.parquet')
df.to_parquet(filePath)